Add technical court documentation

Chralt98 opened the pull request.

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,121 @@ title: Decentralized Court
 
-[aragon court]: https://example.com
-[kleros]: https://example.com
+- court hopping: joining the court just to receive token emission benefits and then exiting the court
+- court pool: all jurors and delegators
+- draw weight: one slash-able `MinJurorStake` that belongs to one vote weight
+- global dispute: a token voting mechanism for all ZTG holders
+- participant: a juror or delegator account inside the court pool
+- vote weight: the derivative voting value of one draw weight; it represents one `MinJurorStake`
```
This needs to be spell-checked. Capitalization at the start of each item, capitalization after the colon and full stop at the end of sentences.

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,121 @@ title: Decentralized Court
+
+The formula to determine the necessary requested vote weight for each appeal is as follows:
+
+`2^(appeal_number) * 31 + 2^(appeal_number) - 1`
```
~~~suggestion
```
2^(appeal_number) * 31 + 2^(appeal_number) - 1
```
~~~

(Or using LaTeX.)

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,121 @@ title: Decentralized Court
+
+### Connection of court to prediction markets
+
+The court serves as a dispute resolution mechanism, which is activated if the market creator specifies it. The market creator also designates the oracle account responsible for reporting the winning outcome. For example, in a soccer game, the market question may be: Which team will win? The outcome tokens could be "A" for the first team winning, "B" for the second team winning, and "DRAW" for a t ...
```
https://github.com/zeitgeistpm/documentation/commit/da8d58860da7f2815e716f828eabaffc37da0b48

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,121 @@ title: Decentralized Court
+The court is inspired by [Kleros][], a project that has
+already experimented with an on-chain court system.
+
+[kleros]: https://kleros.io/
```
https://github.com/zeitgeistpm/documentation/commit/a7d1dc2b5387a4517bd1cfc9fd086044521e5aed

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,121 @@ title: Decentralized Court
+
+`2^(3) * 31 + 2^(3) - 1 = 255`
+
+`255 * 500 ZTG = 127,500 ZTG`
```
https://github.com/zeitgeistpm/documentation/commit/b53c5cabe9b56ea4bf50e917b2f228959f2b6b86

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+dispute is triggered, the court takes over. The court is comprised of jurors and
+delegators who need to lock a certain amount of ZTG tokens to join the court
+system. The more tokens locked, the higher the probability of being selected as
+an active voter (juror) or delegator, who risks funds on behalf of delegated
```
The term _voter_ is only mentioned here once. You seem to be using the term _active juror_ for the rest of the document.

Chralt98 replied:
https://github.com/zeitgeistpm/documentation/pull/81/commits/a19875040e67869b6cc091dad09edd5163eef651

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+jurors. Delegators can transfer their voting rights to active jurors, who
+participate in the voting system. The court uses a plurality voting system,
+meaning the outcome with the most votes wins. Each market is associated with one
+inner court, which can be appealed multiple times if someone believes the
```
The term _inner court_ doesn't seem to have been defined. It seems to be used equivalently to _court case_.

Chralt98 replied:
https://github.com/zeitgeistpm/documentation/pull/81/commits/a19875040e67869b6cc091dad09edd5163eef651

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+participant needs to evoke the `exit_court` dispatch function. The used funds
+(`active_lock`) still remain locked.
+
+### Calculating necessary Draw Weights
```
```suggestion
### Calculating Necessary Draw Weights
```

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+inner court, which can be appealed multiple times if someone believes the
+jurors' decision is unjustified.
+
+### Global Disputes as the last Instance
```
```suggestion
### Global Disputes as the Last Instance
```

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+
+In order to make an appeal, the caller of `appeal` has to reserve a bond. The
+cost of an appeal is calculated as following:
+`cost of appeal = AppealBond * 2^(appeal_number + 1)`
```
```suggestion
`cost of appeal = AppealBond * 2^(appeal_number + 1)`.
```

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+cost of an appeal is calculated as following:
+`cost of appeal = AppealBond * 2^(appeal_number + 1)`
+
+1. first appeal cost: `2000 ZTG (AppealBond) * 2^1 = 4000 ZTG`
+2. second appeal cost: `2000 ZTG (AppealBond) * 2^2 = 8000 ZTG`
+3. third appeal cost: `2000 ZTG (AppealBond) + 2^3 = 16000 ZTG`
```
First letter needs to be capitalized here.

Chralt98 replied:
https://github.com/zeitgeistpm/documentation/pull/81/commits/a19875040e67869b6cc091dad09edd5163eef651

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+court case is triggered by the prediction markets functionality, the court
+requests a small multiple of the constant amount (currently
+`31 * MinJurorStake`) from the total stake of all jurors and delegators
+randomly. The current configuration of `MinJurorStake` is `500 ZTG`. So,
```
The formatting of amounts of currency is inconsistent throughout the document. Sometimes it's in backticks, like here, sometimes its not.

Chralt98 replied:
https://github.com/zeitgeistpm/documentation/pull/81/commits/5063537295985e0db33bc808be8220ca164b2d1c

maltekliemann commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+funds. If the juror or delegator is still actively involved in inner court
+cases, only the unused (non-active) funds are returned. The unused stake is
+equivalent to the stake that was not already selected by the selection algorithm
+(see “The selection algorithm”). In order to exit the court and get the funds
```
The title here needs to be capitalized properly. You could also add a link to the section.

Chralt98 replied:
https://github.com/zeitgeistpm/documentation/pull/81/commits/a19875040e67869b6cc091dad09edd5163eef651

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+dispute is triggered, the court takes over. The court is comprised of jurors and
+delegators who need to lock a certain amount of ZTG tokens to join the court
+system. The more tokens locked, the higher the probability of being selected as
+an active voter (juror) or delegator, who risks funds on behalf of delegated
```
```suggestion
an active juror or delegator, who risks funds on behalf of delegated
```

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+jurors. Delegators can transfer their voting rights to active jurors, who
+participate in the voting system. The court uses a plurality voting system,
+meaning the outcome with the most votes wins. Each market is associated with one
+inner court, which can be appealed multiple times if someone believes the
```
```suggestion
court case, which can be appealed multiple times if someone believes the
```

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+funds. If the juror or delegator is still actively involved in inner court
+cases, only the unused (non-active) funds are returned. The unused stake is
+equivalent to the stake that was not already selected by the selection algorithm
+(see “The selection algorithm”). In order to exit the court and get the funds
```
```suggestion
(see “The Selection Algorithm”). In order to exit the court and get the funds
```

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+
+Assume one specific court case is in the last appeal round (3). The number of
+randomly selected draw weights for jurors and delegators is 255, and, therefore,
+the amount of requested ZTG is 127,500 ZTG.
```
```suggestion
the amount of requested ZTG is `127,500 ZTG`.
```

Chralt98 commented on docs/learn/court.md:
````diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+```
+
+Assume one specific court case is in the last appeal round (3). The number of
+randomly selected draw weights for jurors and delegators is 255, and, therefore,
````
```suggestion
randomly selected draw weights for jurors and delegators is `255`, and, therefore,
```

Chralt98 commented on docs/learn/court.md:
````diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+2^(appeal_number) * 31 + 2^(appeal_number) - 1
+```
+
+Assume one specific court case is in the last appeal round (3). The number of
````
```suggestion
Assume one specific court case is in the last appeal round (`3`). The number of
```

Chralt98 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,298 @@ title: Decentralized Court
+
+Additionally, the court system is incentivized by inflation. Participants who
+stake funds in the court receive newly minted tokens proportional to their
+stake. The current configuration involves a yearly inflation rate of 2% for all
```
```suggestion
stake. The current configuration involves a yearly inflation rate of `2%` for all
```

sea212 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,321 @@ title: Decentralized Court
+
+After the aggregation phase ends and all jurors have had the chance to make
+their votes public, anyone can appeal the plurality decision (during the appeal
+period). This triggers a new court round with more requested jurors and
+delegators' stakes or (in case after the last round) allows a global dispute to
```
```suggestion
period). This triggers a new court round with more requested jurors and total stake or (in case after the last round) allows a global dispute to
```

Chralt98 replied:
Replacing `delegator's` by `total stake` doesn't really fit here. I renamed it to `court participants stakes`.

sea212 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,321 @@ title: Decentralized Court
+
+By making good decisions, you can be rewarded by those who lose. This is done by
+the extrinsic `reassign_court_stakes`. At the end of the appeal phase, the court
+resolves on a winning outcome (see ”Determining the winner outcome”). This
```
```suggestion
resolves on a winning outcome (see [Determining the winner outcome](#determining-the-winner-outcome)). This
```

sea212 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,321 @@ title: Decentralized Court
+resolves on a winning outcome (see ”Determining the winner outcome”). This
+winner outcome is then compared to the jurors voted outcomes. All jurors and
+delegators who sided with a different outcome to the winner outcome get slashed
+according to their draw weights. All jurors, who failed to vote or failed to
+reveal the encrypted vote or got denounced, and their delegators get slashed
+according to their draw weights too. All jurors and their delegators, who sided
+with the winner outcome get the previously mentioned slashed funds proportional
+to their share of all the other winner stake (`total_winner_stake`).
```
```suggestion
according to their draw weights. All jurors who failed to vote or failed to
reveal the encrypted vote or got denounced, as well as their delegators, get also slashed
according to their draw weights. All jurors and their delegators who sided
with the winner outcome get the previously mentioned slashed funds proportional
to their share of all the other winner stake (`total_winner_stake`).
```

sea212 commented on docs/learn/court.md:
```diff
@@ -6,8 +6,321 @@ title: Decentralized Court
+participant if the following condition is true:
+`now - joined_at ≥ InflationPeriod`.
+
+The `YearlyInflation` is able to get set by a `MonetaryGovernanceOrigin` and the
```
```suggestion
The `YearlyInflation` can be configured by a `MonetaryGovernanceOrigin` using the
```

sea212 reviewed (commented):
Pretty solid documentation, good job.

We should maintain some structure in the docs folder imo and not mix everything in `docs/learn`. I propose we move `court_phases.mermaid` and `court_phases.png` into `docs/learn/img`.

Chralt98 commented:
I know it's not so visible now what I have changed, but I indeed addressed all review comments by this https://github.com/zeitgeistpm/documentation/pull/81/commits/7a74c6711912f2587be8b2b15e109f8bbcaff53d.
